src/mcp23017.cpp

```diff
@@ -36,6 +99,20 @@ int Mcp23017::begin(uint8_t directions[8]) {
     int rc;
 
     // TODO: Add device ID check
+    Wire.requestFrom(this->addr, 1);
+    if (Wire.available()){
+        rc = Wire.read();
+    }
+    else{
+        return 0;
```

Comment:
There's a `return 0;` at the end of the function, which implies that 0 means the function passed correctly. However, this `return 0` here implies that the function didn't work correctly. You should standardize what the return values mean and also add a header comment to this function specifying the meanings of the return values.